generators/php/model/src/union/UnionGenerator.ts

```diff
@@ -0,0 +1,1329 @@
+                    argument = php.codeblock(declaredVariableName);
+                }
+                return php.codeblock((writer) => {
+                    writer.write("$value = ");
+                    writer.writeNodeStatement(
+                        php.invokeMethod({
+                            method: "serializeDate",
+                            arguments_: [argument],
+                            static_: true,
+                            on: this.context.getJsonSerializerClassReference()
+                        })
+                    );
+                });
+
+            case "dateTime":
+                if (declaredVariableName != null) {
+                    argument = php.codeblock(declaredVariableName);
+                }
+                return php.codeblock((writer) => {
+                    writer.write("$value = ");
+                    writer.writeNodeStatement(
+                        php.invokeMethod({
+                            method: "serializeDateTime",
+                            arguments_: [argument],
+                            static_: true,
+                            on: this.context.getJsonSerializerClassReference()
+                        })
+                    );
+                });
+
+            case "reference":
+                if (declaredVariableName != null) {
+                    argument = php.codeblock(declaredVariableName);
+                }
+                return php.codeblock((writer) => {
+                    writer.write("$value = ");
+                    writer.writeNodeStatement(
+                        php.invokeMethod({
+                            method: "jsonSerialize",
+                            arguments_: [],
+                            static_: false,
+                            on: argument
+                        })
+                    );
+                });
+
+            case "optional":
+                return php.codeblock((writer) => {
+                    writer.write("$" + asCastMethodName);
+                    writer.write(" = ");
+                    writer.writeNodeStatement(
+                        php.invokeMethod({
+                            method: asCastMethodName,
+                            arguments_: [],
+                            on: php.codeblock("$this")
+                        })
+                    );
+                    writer.controlFlow(
+                        "if",
+                        php.invokeMethod({
+                            method: "is_null",
+                            arguments_: [php.codeblock("$" + asCastMethodName)],
+                            static_: true
+                        })
+                    );
+                    writer.write("$value = ");
+                    writer.writeTextStatement("$" + asCastMethodName);
+                    if (
+                        ["reference", "dateTime", "date", "optional"].includes(type.underlyingType().internalType.type)
```

Comment:
This line made me pause - is there any way we could write a helper here that more clearly describes what this is doing?